assemble/bin/accumulo-service

```diff
@@ -230,14 +231,14 @@ function list_processes() {
     local pid_file
     local pid
     local port
-    pid_file="${ACCUMULO_PID_DIR}/accumulo-${process}.pid"
-    pid=$(<"$pid_file")
-    port=$(netstat -tnlp 2>/dev/null | grep "$pid" | awk '{print $4}' | awk -F ':' '{print $NF}' | paste -sd "," -)
-    # check that only a single port was seen
-    if (($(echo "$port" | grep -c -E '^[0-9]+(,[0-9]+)*$') != 1)); then
-      echo "ERROR unexpected ports $(hostname) process:$process pid:$pid ports:$port" >&2
-    else
+    pid_file="$ACCUMULO_PID_DIR/accumulo-$process.pid"
+    pid=$(<"$pid_file") # read the contents of the file into the $pid variable
+    port=$(ss -tnlp 2>/dev/null | grep -F "pid=$pid," | awk '{print $4}' | awk -F : '{print $NF}' | paste -sd,)
+
+    if [[ $port =~ ^[1-9][0-9]+(,[0-9]+)*$ ]]; then
       echo "$process $pid $port"
+    else
+      echo "ERROR unexpected port format $(hostname) process:$process pid:$pid ports:$port" >&2
```

Comment:
This will echo this to STDERR and then exit normally, without an error code. It should probably do a `return 1` from the function after the echo to ensure the calling code sees that the script experienced an error. This is also happening in a loop, so maybe you want to keep going and exit with an error at the end, instead?

Reply:
That's a good idea. I added a exit code after the loop in b1212f5